### DISCUSSION TITLE
Type for locale (i18n)

### DISCUSSION THREAD

**Discussion by luckasnix** [vercel/next.js#24666] [Ideas]
### Describe the feature you'd like to request

In Next.js 10, we can configure internationalized routing in the "next.config.js" file. In this file we define all locales, but in "getServerSideProps" or "getStaticProps" we receive "locale" as "string". It would be interesting if we received a specific type.

### Describe the solution you'd like

When we define an array of locales, Next.js would use this array to type the locale. For example, if a write the "next.config.js" file in this way:

```javascript
module.exports = {
  i18n: {
    locales: [
      'en-US',
      'pt-BR',
      'es'
    ],
    defaultLocale: 'en-US'
  }
}
```

I'll receive a type like this:

```typescript
type Locale = 'en-US' | 'pt-BR' | 'es'
```

### Describe alternatives you've considered

I don't know if it is possible. It's just an ideia.

**Comment by antoinelin** [comment]
On my side, I wrap the fetched locale from `useRouter` in a `useI18n` hook (on which I get my translated string too) and use type guard to force getting my locale typed as `Locale`.
Same on `getServerSideProps/getStaticProps` I use a type guard before using `locale` to fetch content
